polaris-react/src/components/ResourceList/ResourceList.scss

```diff
@@ -99,6 +99,11 @@ $item-wrapper-loading-height: 64px;
 
 .HeaderWrapper-isSticky {
   box-shadow: var(--p-shadow-md);
+
+  #{$se23} & {
+    box-shadow: var(--p-shadow-xs);
+    border-radius: 0;
```

Comment:
We don't have a token for a 0 border-radius so I'm assuming this is fine for now. 
Though it is odd that we're expecting users to reach outside of the system when explicitly declaring no border-radius. Here's a related [support question](https://shopify.slack.com/archives/C4Y8N30KD/p1688497417492439)

cc @lgriffee  there's probably context I'm missing 🙏